### PR TITLE
Add additional example on tag filters for workflows

### PR DESCRIPTION
**Pull request by nbialostosky**
# Description
This PR adds an additional example of how to utilize tag and branch filtering to run specific workflows.

# Reasons
We had a customer make the suggestion of adding this example to the documentation and I believe it adds some additional clarity to the different ways you can use tags to filter workflows.

### CODE REVIEW

File: jekyll/_cci2/workflows.md [hunk 1]
```diff
@@ -430,6 +430,48 @@ workflows:
               ignore: /.*/
 ```
 
+In the example below, two jobs are defined (`test` and `deploy`) and three workflows utilize those jobs:
+
+- The `build` workflow runs for all branches except `main` and is not run on tags.
```
Comment (rosieyohannan) [inline jekyll/_cci2/workflows.md:435]: For this to be true do we need to add the `tags:` `ignore:` keys for the `build` workflow?

Comment (nbialostosky) [reply jekyll/_cci2/workflows.md:435]: @rosieyohannan great question! You actually don't since we don't automatically run workflows on tags, so by not including it doesn't run on tags. I set up this config example in one of my test accounts, and you can see the three scenarios here:

Screen Shot 2020-10-15 at 8 06 26 AM

Comment (rosieyohannan) [reply jekyll/_cci2/workflows.md:435]: Ah interesting! Thanks :-)

File: jekyll/_cci2/workflows.md [hunk 2]
```diff
@@ -430,6 +430,48 @@ workflows:
...
+          filters: &filters-staging
+            branches:
+              only: main
+            tags:
+              ignore: /.*/
+      - deploy:
+          requires:
+            - build
+          filters:
+            <<: *filters-staging
```
Comment (rosieyohannan) [inline jekyll/_cci2/workflows.md:459]: Can we just add comments in for these lines to say we've using the same 'staging' filters for this job (same for use of alias in production), just to spell it out?

**Review by rosieyohannan — APPROVED** [review]
🎉

File: jekyll/_cci2/workflows.md (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/workflows.md:435]: Maybe we can add a comments to say this explicitly if it's not too wordy?

File: jekyll/_cci2/workflows.md (hunk 2 above)
Comment (nbialostosky) [reply jekyll/_cci2/workflows.md]: I updated this with commit 6502576 can you check to see if that works?

Comment (rosieyohannan) [reply jekyll/_cci2/workflows.md]: Great, thank you!!
